[Adam Ang Zi Jun] iP

adammangzijun opened the pull request:
# Adam Chat Bot

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

### Adam Chat Bot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

### All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :smiley:
## And it is **FREE**!

### Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

adammangzijun committed:
Feature 0: added the base message

adammangzijun committed:
Feature 1: added user input and cases

adammangzijun committed:
Feature 2: added add function and list function

adammangzijun committed:
Feature 3: added Task class and mark/unmark function

adammangzijun committed:
Feature 4: added subclasses of Task

adammangzijun committed:
Feature Test: Trying out test cases

adammangzijun committed:
Feature 5: handled the error of inputs for todo, deadline, event and also invalid command inputs

adammangzijun committed:
Feature 6: added delete function and changed String[] array to ArrayList<Task>.

adammangzijun committed:
Feature 6: Updated on the documentations and edited the formatting of code

adammangzijun committed:
Feature 6: updated the error handling for delete

adammangzijun committed:
Add enum and modified javaDoc

adammangzijun committed:
added local storage of data, without handling corrupted data

adammangzijun committed:
added date and time for events and deadline

adammangzijun committed:
Merge branch 'branch-Level-7'

adammangzijun committed:
Added more OOP

adammangzijun committed:
Merge branch 'add-gradle-support' into branch-Level-7

adammangzijun committed:
Merge commit '28ad2b878f1a21d03879e81cd441d0175d89e6b1' into HEAD

adammangzijun committed:
Added gradle, packages and JUnit

adammangzijun committed:
Changed build.gradle and also added duke.Jar

adammangzijun committed:
Merge commit 'bd902eb05a9ddd23481a7996ca44e8610d57705e'

# Conflicts:
#	build.gradle

cmHuang777 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,185 @@
+                return new ByeCommand();
+            case "list" :
+                return new ListCommand();
+            case "todo" :
```
Should the switch case be at the same level of indentation according to the CS2103 style guide?

cmHuang777 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,185 @@
+                    // add item into list
+                    Deadline task = new Deadline(taskName, dueDate);
+                    return new AddCommand(task);
+                } catch ( ArrayIndexOutOfBoundsException e) {
```
Perhaps there should not be any spacing between the ( and A ?

cmHuang777 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,185 @@
+                    if (input.split(" ").length == 1) {
+                        return new ErrorCommand("☹ OOPS!!! The description of a event cannot be empty");
+                    } else {
+                        return new ErrorCommand("☹ OOPS!!! The format of starting date or due date is wrong, please write it in the format correct format");
```
Perhaps this line is too long? It seems to be exceeding the character limit of 120 per line according to the style guide.
May be you want to consider breaking it up?

cmHuang777 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,185 @@
+                    String dueDateInput = input.split(" /from | /to ", 3)[2];
+
+                    // ArrayList containing possible formats of Date and Time.
+                    List<String> formatStrings = Arrays.asList(
```
May be the variable can be better named? For example, name it as stringFormat to make it sound like a thing, instead of a action verb.

cmHuang777 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+        this.path = Paths.get(this.filepath);
+    }
+    /**
+     * Create the directory and file to store the previous list, else update it to the current list.
```
Perhaps the start of javadoc should be a verb with s behind according to the style guide?
eg, Creates instead of Create

cmHuang777 reviewed (commented):
Overall looks good!
Just some comments about style.

pzl111 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+     *
+     * @return The ArrayList of tasks saved from previous usage.
+     */
+
+
```
According to coding standards, perhaps there should not be empty lines below javadocs and the method declaration?

pzl111 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,185 @@
+                        return new ErrorCommand("☹ OOPS!!! The format of starting date or due date is wrong, please write it in the format correct format");
+                    }
+                } catch (DateTimeParseException e) {
+                    return new ErrorCommand("☹ OOPS!!! The format of starting date or due date is wrong, please write it in the format correct format");
```
This line looks to be over 120 characters, maybe breaking it up into another line would be better?

pzl111 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+        }
+    }
+
+    /**
+     * Update the list of tasks in the duke.txt according to the current list
+     *
+     * @param tasks the ArrayList of tasks that were added
+     */
```
I think the sentences in javadocs should end with a full stop?

pzl111 commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,76 @@
+    }
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + startDate + " to: " + dueDate + ")";
```
Good use of spacing between the operators "+"

pzl111 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,46 @@
+    private boolean isDone = false;
+    private TaskType taskType;
+
+    //constructor to create a Task object with a task description
```
I think javadocs should always start with /** as the comment, rather than //, regardless if its a single or multiline comment?

pzl111 reviewed (commented):
Overall good coding standards followed, just some minor mistakes here and there.

adammangzijun committed:
Added javaDocs

adammangzijun committed:
Change styling

adammangzijun committed:
Add Find function

adammangzijun committed:
Merge branch 'branch-A-CodingStandard'

adammangzijun committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/duke.txt
#	src/main/java/parser/Parser.java
#	src/main/java/ui/Ui.java

adammangzijun committed:
Check style with CheckStyle gradle
Updated most file, mostly the imports as well as small details like indentations and white spaces.

adammangzijun committed:
Add GUI, as well as added duke package and GUI package

adammangzijun committed:
Merge branch 'branch-Level-10'

adammangzijun committed:
Parser class: Add assert statements

Assertions are added to the parse method to ensure that
user inputs are not empty and that there's no issues
when creating new tasks.

adammangzijun committed:
Change coding quality and add JavaDoc

Parser class: parse method was very long, violating both
the long methods and deep nesting rules. Also included the JavaDocs.

MainWindow class: Added some JavaDoc as well.

adammangzijun committed:
Add update feature

Issues indentified so far:
1.Unable to update field due to wrong "field" input but no error
message.

2.Error for not being able to update task not handled.

adammangzijun committed:
Merge pull request #2 from adammangzijun/heads/A-Assertions

Heads/a assertions

adammangzijun committed:
Merge branch 'master' into heads/A-CodeQuality

adammangzijun committed:
Merge pull request #3 from adammangzijun/heads/A-CodeQuality

Change coding quality and add JavaDoc

adammangzijun committed:
Merge branch 'master' into heads/BCD-Extension

adammangzijun committed:
Merge pull request #4 from adammangzijun/heads/BCD-Extension

Add update feature

adammangzijun committed:
Update minor checkstyles and delete duke.datd.txt file

adammangzijun committed:
Add README and change GUI

adammangzijun committed:
Create new jar and release

adammangzijun committed:
Change name of image from UI.png to Ui.png
